toast-tk-runtime/src/main/java/io/toast/tk/runtime/block/ArgumentBlockHelper.java

```diff
@@ -0,0 +1,117 @@
+package io.toast.tk.runtime.block;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Type;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class ArgumentBlockHelper {
+
+	private static final Logger LOG = LogManager.getLogger(ArgumentBlockHelper.class);
+
+	private ArgumentBlockHelper() {
+		
+	}
+	
+	public static boolean isArgumentFailureMessage(Exception error, Parameter[] params, Object[] inputArgList) {
+		if(!("argument type mismatch".equals(error.getMessage())) 
+				&& !(error instanceof InvocationTargetException && error.getCause() instanceof ClassCastException)) {
+			return false;
+		}
+		
+		String inputArg = "";
+		try {
+			for(Object input : inputArgList) {
+				inputArg = writeMessage(inputArg, input);
+			}
+		} catch (Exception e) {
+			LOG.error(e.getMessage(), e);
+		}
+		
+		String arg = "";
+		for(Parameter param : params) {
+			arg = writeMessage(arg, param.getParameterizedType());
+		}
+
+		String[] inputArgs = inputArg.split(System.lineSeparator());
+		String[] args = arg.split(System.lineSeparator());
+		
+		if(inputArgs.length != args.length) {
+			return true;
+		}
+		
+		for(int i = 0; i < inputArgs.length; i ++) {
+			String inputTemp = inputArgs[i];
+			String argTemp = args[i];
+			if(!inputTemp.equals(argTemp)) {
+				return true;
+			}
+		}
+		
+		return false;
+	}
+	
+	public static String getArgumentFailureMessage(Parameter[] params, 
+			Object[] inputArgList) {
+		String res = "Your inputs arguments are :" + System.lineSeparator();
+		try {
+			String inputArg = "";
+			for(Object input : inputArgList) {
+				inputArg = writeMessage(inputArg, input);
+			}
+			res += inputArg;
+		} catch (Exception e) {
+			LOG.error(e.getMessage(), e);
+		}
+		
+		res += System.lineSeparator() + "Instead of :" + System.lineSeparator();
+
+		String args = "";
+		for(Parameter param : params) {
+			args = writeMessage(args, param.getParameterizedType());
+		}
+		res += args;
+		
+		return res;
+	}
+	
+	@SuppressWarnings("rawtypes")
+	private static String writeMessage(String args, Object input) {
+		String className = input.getClass().getName().trim();
+		if(input instanceof List && ((List) input).size() != 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)